### PR TITLE
Add/exclude

### PR DESCRIPTION
**Pull request by amitgilad3**

### CODE REVIEW

File: src/api/consumer/lib/add.js [hunk 1]
```diff
@@ -42,6 +42,35 @@ export default async function addAction(componentPaths: string[], id?: string, m
       return { id: componentId.toString(), files };
     };
 
+    const getValidBitId = (box: string, name: string): BitId => {
```
Comment (GiladShoham) [inline src/api/consumer/lib/add.js]: Remove this function, it's now a static function in BitId class.. use it from there

File: e2e/commands/add.e2e.js [hunk 2]
```diff
@@ -57,4 +65,62 @@ describe('bit add command', function () {
       expect(bitMap).to.have.property('test/foo2');
     });
   });
+  describe('add component with exclude', () => {
```
Comment (GiladShoham) [inline e2e/commands/add.e2e.js]: please add one test with few excluded with , between them

Comment (amitgilad3) [inline e2e/commands/add.e2e.js]: done

File: src/api/consumer/lib/add.js (hunk 1 above)
Comment (amitgilad3) [inline src/api/consumer/lib/add.js]: done

File: src/api/consumer/lib/add.js [hunk 3]
```diff
@@ -42,6 +42,28 @@ export default async function addAction(componentPaths: string[], id?: string, m
...
+          const relativeComponentPath = getPathRelativeToProjectRoot(componentPath, consumer.getPath());
+          const matches = await glob(path.join(relativeComponentPath, '**'), { cwd: consumer.getPath(), nodir: true });
+          matches.forEach((match) =>  files[match] = match);
+        } else { // is file
+          const relativeFilePath = getPathRelativeToProjectRoot(componentPath, consumer.getPath());
+          files[relativeFilePath] = relativeFilePath
+        }
+        return files;
+      }));
+      return filesArr.reduce((acc, x) => {
```
Comment (GiladShoham) [inline src/api/consumer/lib/add.js]: Use mergeAll from ramda instead

File: src/api/consumer/lib/add.js [hunk 4]
```diff
@@ -42,6 +42,28 @@ export default async function addAction(componentPaths: string[], id?: string, m
       return { id: componentId.toString(), files };
     };
 
+    async function getAllFiles(excluded){
```
Comment (GiladShoham) [inline src/api/consumer/lib/add.js]: change excluded name (also inside the function)

File: src/api/consumer/lib/add.js [hunk 5]
```diff
@@ -99,13 +121,25 @@ export default async function addAction(componentPaths: string[], id?: string, m
...
     });
 
-    const mapValues = await Promise.all(mapValuesP);
+    var mapValues = await Promise.all(mapValuesP);
+
+    if (exclude){
+      const resolvedExcludedFiles =  await getAllFiles(exclude);
+      mapValues.forEach(mapVal => {
+        mapVal.files=mapVal.files.filter(key => !resolvedExcludedFiles[key.relativePath] );
+        mapVal.testsFiles = mapVal.testsFiles.filter(x=>!resolvedExcludedFiles[x])
```
Comment (GiladShoham) [inline src/api/consumer/lib/add.js]: change x to meaningful name

**Pull request merged**